src/auth/client.ts

```diff
@@ -26,6 +26,10 @@ export interface AuthenticateServerOptions extends AbortOptions {
   hostname?: string
 }
 
+export interface VerifyPeerOptions {
+  verifyPeer?(peerId: PeerId, options: AbortOptions): Promise<boolean>
```

Comment:
Sorry, one last suggestion to make the async optional:

```suggestion
  verifyPeer?(peerId: PeerId, options: AbortOptions): Promise<boolean> | boolean
```

Reply:
I didn't know you could do that. That's interesting